src/DocTest/Parser.elm

```diff
@@ -1,144 +1,143 @@
-module DocTest.Parser
-    exposing
-        ( parse
-        , parseImports
-        , parseComments
-        )
+module DocTest.Parser exposing (parse)
 
-import String
-import Regex exposing (..)
-import List.Extra
 import DocTest.Types exposing (..)
+import List.Extra
+import Regex exposing (Regex, HowMany(..))
+import String
 
 
 parse : String -> TestSuite
 parse str =
     { imports = parseImports str
     , tests =
-        str
-            |> parseComments
-            |> List.Extra.groupWhile (\x y -> not <| assertionPrefix y)
-            |> List.map (toTest ( [], [] ) "")
+        parseComments str
+            |> List.concatMap (filterNotDocTest << List.filterMap parseDocTest << String.lines)
+            |> List.Extra.groupWhile (\x y -> not <| isAssertion y)
+            |> List.filterMap toTest
     }
 
 
-toTest : ( List String, List String ) -> String -> List String -> Test
-toTest ( assertion, expectation ) last str =
-    case str of
-        h :: rest ->
-            if assertionPrefix h then
-                toTest ( h :: assertion, expectation ) h rest
-            else if continuationPrefix h then
-                toTest ( h :: assertion, expectation ) last rest
-            else if assertionPrefix last then
-                toTest ( assertion, h :: expectation ) last rest
-            else
-                toTest ( assertion, expectation ) "" rest
+parseImports : String -> List String
+parseImports str =
+    Regex.find All importRegex str
+        |> List.map .match
 
-        _ ->
-            let
-                clean x =
-                    List.map replacePrefix x
-                        |> List.reverse
-                        |> String.join " "
-            in
-                Test (clean assertion) (clean expectation)
 
+parseComments : String -> List String
+parseComments str =
+    Regex.find All commentRegex str
+        |> List.map .match
 
-parseImports : String -> List String
-parseImports str =
-    str
-        |> String.lines
-        |> List.foldl parseImport []
-        |> List.reverse
 
+parseDocTest : String -> Maybe Syntax
+parseDocTest =
+    oneOf
+        [ choice assertionRegex Assertion
+        , choice continuationRegex Continuation
+        , choice expectationRegex Expectation
+        ]
 
-parseImport : String -> List String -> List String
-parseImport str acc =
-    if contains (regex "^import\\s.*") str then
-        str :: acc
-    else
-        acc
 
+importRegex : Regex
+importRegex =
+    Regex.regex "import\\s.*"
```

Comment:
I added a stopgap solution for imports. I think we should explicitly import modules (syntax or config json).